src/app/clusters/operational-state-server/operational-state-server.cpp

```diff
@@ -0,0 +1,417 @@
+
+    // set operational error
+    SetOperationalState(currentOperationalState);
+
+    // copy current operational error
+    operationalError = currentErrState;
+
+    // return operational error
+    response.commandResponseState.errorStateID = static_cast<OperationalState::ErrorStateEnum>(operationalError.ErrorStateID);
+    response.commandResponseState.errorStateLabel.SetNonNull(operationalError.ErrorStateLabel,
+                                                             strlen(operationalError.ErrorStateLabel));
+    response.commandResponseState.errorStateDetails.Emplace(operationalError.ErrorStateDetails,
+                                                            strlen(operationalError.ErrorStateDetails));
+
+    ctx.mCommandHandler.AddResponse(ctx.mRequestPath, response);
+}
+
+void OperationalStateServer::HandleResumeState(HandlerContext & ctx, const Commands::Resume::DecodableType & req)
+{
+    ChipLogDetail(Zcl, "OperationalState: HandleResumeState");
+    Commands::OperationalCommandResponse::Type response;
+    OperationalErrorStateStruct currentErrState;
+    OperationalStateStruct currentOperationalState;
+
+    // copy current operational state
+    currentOperationalState = operationalState;
+    // copy current operational error
+    currentErrState = operationalError;
+
+    // callback
+    delegate->HandleResumeState(currentOperationalState, currentErrState);
+
+    // set operational error
+    SetOperationalState(currentOperationalState);
+
+    // copy current operational error
+    operationalError = currentErrState;
+
+    // return operational error
+    response.commandResponseState.errorStateID = static_cast<OperationalState::ErrorStateEnum>(operationalError.ErrorStateID);
+    response.commandResponseState.errorStateLabel.SetNonNull(operationalError.ErrorStateLabel,
+                                                             strlen(operationalError.ErrorStateLabel));
+    response.commandResponseState.errorStateDetails.Emplace(operationalError.ErrorStateDetails,
+                                                            strlen(operationalError.ErrorStateDetails));
+
+    ctx.mCommandHandler.AddResponse(ctx.mRequestPath, response);
+}
+
+void OperationalStateServer::HandleStartState(HandlerContext & ctx, const Commands::Start::DecodableType & req)
+{
+    ChipLogDetail(Zcl, "OperationalState: HandleStartState");
+    Commands::OperationalCommandResponse::Type response;
+    OperationalErrorStateStruct currentErrState;
+    OperationalStateStruct currentOperationalState;
+
+    // copy current operational state
+    currentOperationalState = operationalState;
+    // copy current operational error
+    currentErrState = operationalError;
+
+    // callback
+    delegate->HandleStartState(currentOperationalState, currentErrState);
+
+    // set operational error
+    SetOperationalState(currentOperationalState);
+
+    // copy current operational error
+    operationalError = currentErrState;
```

Comment:
It's not clear to me from the spec whether the OperationalError stored in the attribute is supposed to be updated on every command...

For example, for the Pause command the spec says:

> If this command is received when already in the Paused state the device SHALL respond with an OperationalCommandResponse command with an ErrorStateID of NoError but take no further action.

> A device that is unable to honour the Pause command for whatever reason SHALL respond with an OperationalCommandResponse command with an ErrorStateID of CommandInvalidInState but take no further action.

and I would not expect either one to update the error state attribute, since nothing changed about the state.

Furthermore, all these command impls _should_ be implementing the spec-required business logic (like that bit about receiving Pause when in Paused state) and they don't seem to be.